interfaces.go

```diff
@@ -8,22 +8,27 @@ package optimize
 //
 // It uses a reverse-communication interface between the optimization method
 // and the caller. Method acts as a client that asks the caller to perform
-// needed operations via RequestType returned from Init and Iterate methods.
+// needed operations via Operation returned from Init and Iterate methods.
 // This provides independence of the optimization algorithm on user-supplied
 // data and their representation, and enables automation of common operations
 // like checking for (various types of) convergence and maintaining statistics.
 //
 // A Method can command an Evaluation, a MajorIteration or NoOperation operations.
+//
 // An evaluation operation is one or more of the Evaluation operations
 // (FuncEvaluation, GradEvaluation, etc.) which can be combined with
 // the bitwise or operator. In an evaluation operation, the requested routines
 // will be evaluated at the point specified in Location.X. The corresponding
 // fields of Location will be filled with the results from the routine and can
-// be retrieved upon the next call to Iterate. Alternatively, a Method can
-// declare a MajorIteration. In a MajorIteration, all values in Location must
-// be valid and consistent, and are interpreted as a new minimum. Convergence
-// of the optimization (GradientThreshold, etc.) will be checked using this new
-// minimum.
+// be retrieved upon the next call to Iterate. The caller must not modify
+// the contents of Location between calls to Init and Iterate apart from the
+// commanded evaluations so that a Method can use it as a storage for gradually
+// building a valid location.
+//
+// Alternatively, a Method can declare a MajorIteration. In a MajorIteration,
```

Comment:
Instead of an Evaluation type, a method may declare MajorIteration. In a MajorIteration, the values in the fields of Location are treated as a potential optimizer. The convergence of the optimization routine (GradientThreshold, etc.) is checked at this new best point. In a MajorIteration, the fields of Location must be valid and consistent.